### DISCUSSION TITLE
Set up proportions of each colour

### DISCUSSION THREAD

**Discussion by yiqi171717** [koenderks/aRtsy#159] [Q&A] [answered]
Hi Koen, I'm trying your package to generate some cover pictures for our school newsletter and I wonder if it's possible at all to set the proportion of each colour in the artwork? We did a survey of everyone's favourite colour and would like to turn these statistics into some art :) 
Thanks a lot!

**Comment by koenderks** [comment, marked as answer]
Hi @yiqi171717,

Sounds like a cool idea! I think a good attempt would be to first create a vector of colors that has the appropriate proportions in it, like is shown below. In the vector `colors`, the color `blue` occurs (approximately) 30%, `red` occurs 20%, and `yellow` occurs 50% of the times.
```
set.seed(1)
colors <- sample(c("blue", "red", "yellow"), size = 1000, prob = c(0.3, 0.2, 0.5), replace = TRUE)
```
Next, you can use this color vector in one of the functions. For example:
```
aRtsy::canvas_flow(colors = colors, lwd = 0.5)
```
![hoi](https://user-images.githubusercontent.com/25059399/140803063-3099664d-6a91-458a-a2f2-abffce3d7475.png)

Hope this helps! :)

**Reply by yiqi171717** [reply]
Hi @koenderks !
Thanks a lot for your rapid reply! Does your function take in hex codes? Because I already tried something similar to your suggestion: I cleaned up the survey results, pulled the hex code column as a vector and used that as the input colour palette. However that just generated an error :/

**Reply by koenderks** [reply]
Oh hmm that is unusual... I believe it should also work with hex codes. For example,

```
set.seed(1)
colors <- sample(c("#F4EFD6", "#267029", "#422670"), size = 1000, prob = c(0.3, 0.2, 0.5), replace = TRUE)
aRtsy::canvas_flow(colors = colors, lwd = 0.5)
```
gives me:
![hoi](https://user-images.githubusercontent.com/25059399/140805935-26e55a49-4679-41bc-b79d-7690e0d84866.png)

Which function are you using, and what is the error message you get?

**Reply by yiqi171717** [reply]
So I have a data frame, colours.df, with 50 rows (50 answers). One column is the name of the colour and the other is the hex code (named "hex").

hex <- pull(colours.df, hex)
artwork <- canvas_strokes(colors = hex)

it gave me almost instantly an error message "Error in draw_strokes(X = canvas, neighbors = neighborsLocations, s = length(colors),  : 
  function 'Rcpp_precious_remove' not provided by package 'Rcpp' "

**Reply by koenderks** [reply]
It seems an issue with the version of 'Rcpp'. Maybe updating the package using 'install.packages("Rcpp")' will solve this problem!

**Reply by yiqi171717** [reply]
you're right, I updated Rcpp and it all works fine now! Thank you so much for your help, it's such a fun package to use :D

**Comment by yiqi171717** [comment]
[Issue11_cover_watercolour.pdf](https://github.com/koenderks/aRtsy/files/7626662/Issue11_cover_watercolour.pdf)
the end product ;) (proper citation follows inside the issue)
thanks again for all your help!

**Reply by koenderks** [reply]
Awesome, this looks really cool and makes my day! Thank you for sharing :)
